src/store/selectors/solanaWallet.ts

```diff
@@ -81,16 +89,58 @@ export const swaplinePairs = createSelector(
           ...allSynthetics[swapline.synthetic.toString()],
           balance: syntheticAccount ? syntheticAccount.balance : new BN(0)
         },
-        collateralData: allCollaterals[swapline.collateral.toString()].symbol !== 'WSOL'
-          ? {
-            ...allAssets[allCollaterals[swapline.collateral.toString()].assetIndex],
-            ...allCollaterals[swapline.collateral.toString()],
-            balance: collateralAccount ? collateralAccount.balance : new BN(0)
-          } : {
-            ...allAssets[allCollaterals[swapline.collateral.toString()].assetIndex],
-            ...allCollaterals[swapline.collateral.toString()],
-            balance: wSOLBalance
-          }
+        collateralData:
+          allCollaterals[swapline.collateral.toString()].symbol !== 'WSOL'
+            ? {
+              ...allAssets[allCollaterals[swapline.collateral.toString()].assetIndex],
+              ...allCollaterals[swapline.collateral.toString()],
+              balance: collateralAccount ? collateralAccount.balance : new BN(0)
+            }
+            : {
+              ...allAssets[allCollaterals[swapline.collateral.toString()].assetIndex],
+              ...allCollaterals[swapline.collateral.toString()],
+              balance: wSOLBalance
+            }
+      }
+
+      return pair
+    })
+  }
+)
+export interface BorrowedPair extends Vault {
+  collateralData: ExchangeCollateralTokens
+  syntheticData: ExchangeSyntheticTokens
+}
+export const vaultPairs = createSelector(
+  vaults,
+  synthetics,
+  collaterals,
+  accounts,
+  assets,
+  balance,
+  (allVaults, allSynthetics, allCollaterals, tokensAccounts, allAssets, wSOLBalance) => {
+    return Object.values(allVaults).map(vault => {
+      const syntheticAccount = tokensAccounts[vault.synthetic.toString()]
+      const collateralAccount = tokensAccounts[vault.collateral.toString()]
+      const pair: BorrowedPair = {
+        ...vault,
+        syntheticData: {
+          ...allAssets[allSynthetics[vault.synthetic.toString()].assetIndex],
+          ...allSynthetics[vault.synthetic.toString()],
+          balance: syntheticAccount ? syntheticAccount.balance : new BN(0)
+        },
+        collateralData:
+          allCollaterals[vault.collateral.toString()].symbol !== 'WSOL'
+            ? {
+              ...allAssets[allCollaterals[vault.collateral.toString()].assetIndex],
+              ...allCollaterals[vault.collateral.toString()],
+              balance: collateralAccount ? collateralAccount.balance : new BN(0)
+            }
+            : {
+              ...allAssets[allCollaterals[vault.collateral.toString()].assetIndex],
+              ...allCollaterals[vault.collateral.toString()],
+              balance: wSOLBalance
+            }
```

Comment:
I dont get it. Why mix vault data with user data ? This is separate information and you are creating macro selector. Selectors should be as small as possible ! Why you need `balance` in `vaultPairs` ? your balance is not part of vault is external information. If you create this kind of selectors dont call them `vaultPairs` because its just not true. Its more `usersPairs`